add new cache page (#9949)
getsentry/sentry-docs: docs/product/performance/cache-monitoring/index.mdx

---
title: "Cache Monitoring"
sidebar_order: 50

> description: "Learn more about Sentry's Cache monitoring capabilities and how it can help improve your application's performance."

Author: ```suggestion
description: "Learn more about cache monitoring with Sentry and how it can help improve your application's performance."
```

---

<Include name="feature-stage-alpha.mdx" />

> A cache can be used to speed up data retrieval, thereby improving application performance. Because instead of getting data from a potentially slow data layer, your application will be getting data from, *in a best case scenarios*, memory instead. Caching can speed up read-heavy workloads for applications like Q&A portals, gaming, media sharing, and social networking.
> 

Author: ```suggestion
A cache can be used to speed up data retrieval, thereby improving application performance. Because instead of getting data from a potentially slow data layer, your application will be getting data from, (in a best case scenario), memory instead. Caching can speed up read-heavy workloads for applications like Q&A portals, gaming, media sharing, and social networking.

```

A successful cache results in a high hit rate which means the data was present when fetched. A cache miss occurs when the data fetched was not present in the cache. If you have performance monitoring enabled and your application uses caching, you can see how your caches are performing with Sentry.

Sentry's cache monitoring provides insights into cache utilization and latency so that you can improve the performance of the endpoints that interact with caches.

> Starting with the [Cache page](https://sentry.io/orgredirect/organizations/:orgslug/performance/cache/), you get an overview of the transactions within your application that are making at least one lookup against a cache. From there, you can dig into specific cache span operations by clicking a transaction and viewing its sample list.

Author: ```suggestion
Starting with the [Cache page](/orgredirect/organizations/:orgslug/performance/cache/), you get an overview of the transactions within your application that are making at least one lookup against a cache. From there, you can dig into specific cache span operations by clicking a transaction and viewing its sample list.
```

### Prerequisites and Limitations

Cache monitoring currently supports [auto instrumentation](/platform-redirect/?next=%2Fperformance%2Finstrumentation%2Fautomatic-instrumentation) for [Django's cache framework](https://docs.djangoproject.com/en/5.0/topics/cache/). Other frameworks, such as [Node.js' redis](https://www.npmjs.com/package/redis), require opt-in or custom instrumentation.

> ### Custom and Opt-in instrumentation

Reviewer: I think its fine to have this seperate tbh, although maybe we should change the titles to `Automatic Instrumentation` and `Custom Instrumentation`.

I don't believe `opt-in` instrumentation needs to be mentioned, this would fall under `automatic` as your opting in to automatic instrumentation.

> ### Custom and Opt-in instrumentation

Reviewer: ```suggestion
### Custom instrumentation
```

> If available, opt-in instrumentation is documented on an environment-by-environment basis as listed below:
> 
> - [Python SDK](/platforms/python/performance/instrumentation/custom-instrumentation/)
> - JavaScript SDKs - [NEW PAGE JavaScript → Setup Performance → Cache]

Reviewer: ```suggestion
If available, customer instrumentation is documented on an environment-by-environment basis as listed below:

- [Python SDK](/platforms/python/performance/instrumentation/custom-instrumentation/cache-module/)
- [JavaScript SDKs] - (/platforms/javascript/performance/instrumentation/custom-instrumentation/cache-module/)
```

Python link should be good, javascript link is in PR https://github.com/getsentry/sentry-docs/pull/10099

> - [Python SDK](https://docs.sentry.io/platforms/python/performance/instrumentation/custom-instrumentation/)
> - JavaScript SDKs - [NEW PAGE JavaScript → Setup Performance → Cache]
> - [Laravel SDK](https://docs.sentry.io/platforms/php/guides/laravel/performance/instrumentation/custom-instrumentation/)
> - [Java SDK](https://docs.sentry.io/platforms/java/distributed-tracing/custom-instrumentation/)
> - [Ruby SDK](https://docs.sentry.io/platforms/ruby/)
> - [.NET SDK](https://docs.sentry.io/platforms/dotnet/distributed-tracing/custom-instrumentation/)
> - [Symfony SDK](https://docs.sentry.io/platforms/php/guides/symfony/distributed-tracing/custom-instrumentation/)

Author: ```suggestion
- [Python SDK](/platforms/python/performance/instrumentation/custom-instrumentation/)
- JavaScript SDKs - [NEW PAGE JavaScript → Setup Performance → Cache]
- [Laravel SDK](/platforms/php/guides/laravel/performance/instrumentation/custom-instrumentation/)
- [Java SDK](/platforms/java/distributed-tracing/custom-instrumentation/)
- [Ruby SDK](/platforms/ruby/)
- [.NET SDK](/platforms/dotnet/distributed-tracing/custom-instrumentation/)
- [Symfony SDK](/platforms/php/guides/symfony/distributed-tracing/custom-instrumentation/)
```
